### ISSUE TITLE
Failure of the lookups integration test

### ISSUE THREAD

**Issue by pcahyna** [ansible/ansible#28405]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

integration test for lookups

##### ANSIBLE VERSION

```
ansible 2.3.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, May  3 2017, 07:55:04) [GCC 4.8.5 20150623 (Red Hat 4.8.5-14)]
```

##### OS / ENVIRONMENT

RHEL 7.4
##### SUMMARY

The "lookups" integration test (test/integration/targets/lookups/tasks/main.yml) can fail when the first environment variable has a value of "false" (or anything else which is interpreted as boolean). The set_fact module will convert the result of the env lookup to boolean, so the value will be false, but the result from the shell command (known_var_value.stdout) will be string, so "false". The two values are not equal.

Seems to be fixed in the master branch in cd22f67154be89d5760c6c3e2f6e8a1ea3f358f6.
##### STEPS TO REPRODUCE

```
test/runner/ansible-test integration -vvv posix/ci/ --start-at lookups
```

##### EXPECTED RESULTS

the test should succeed.
##### ACTUAL RESULTS

```
TASK [lookups : use env lookup to get known var] *******************************
task path: /root/rpmbuild/BUILD/ansible-2.3.1.0/test/integration/targets/lookups/tasks/main.yml:160
ok: [testhost] => {
    "ansible_facts": {
        "test_val": false
    }, 
    "changed": false
}

TASK [lookups : debug] *********************************************************
task path: /root/rpmbuild/BUILD/ansible-2.3.1.0/test/integration/targets/lookups/tasks/main.yml:164
ok: [testhost] => {
    "known_var_name.stdout": "ANSIBLE_HOST_KEY_CHECKING"
}

TASK [lookups : debug] *********************************************************
task path: /root/rpmbuild/BUILD/ansible-2.3.1.0/test/integration/targets/lookups/tasks/main.yml:165
ok: [testhost] => {
    "known_var_value.stdout": "false"
}

TASK [lookups : debug] *********************************************************
task path: /root/rpmbuild/BUILD/ansible-2.3.1.0/test/integration/targets/lookups/tasks/main.yml:166
ok: [testhost] => {
    "test_val": false
}

TASK [lookups : compare values] ************************************************
task path: /root/rpmbuild/BUILD/ansible-2.3.1.0/test/integration/targets/lookups/tasks/main.yml:168
fatal: [testhost]: FAILED! => {
    "assertion": "test_val == known_var_value.stdout", 
    "changed": false, 
    "evaluated_to": false, 
    "failed": true
}
```

**Comment by bcoca** [comment]
fixed in cd22f67

**Comment by pcahyna** [comment]
I know it is fixed there, but could it be backported to the 2.3 stable branch? I can prepare a PR.
